app/code/Magento/Indexer/Setup/RecurringData.php

```diff
@@ -49,7 +49,8 @@ public function install(ModuleDataSetupInterface $setup, ModuleContextInterface
         foreach (array_keys($this->configInterface->getIndexers()) as $indexerId) {
             $indexer = $this->indexerFactory->create()->load($indexerId);
             if ($indexer->isScheduled()) {
-                $indexer->getView()->unsubscribe()->subscribe();
+                $indexer->setScheduled(false);
```

Comment:
Please test that this not cause change log cleanup

Reply:
Yes, this will cause the changelog tables to be removed as it will call `$indexer->getView()->unsubscribe()` internally. Please can you help me understand the purpose of this recurring setup task if that behaviour is not desired?

Reply:
Further to this, I have discovered (and documented in a code comment) the purpose of this line. I've preserved the behaviour here - no changelog tables will be cleaned up when running this recurring setup script.